src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -217,9 +269,20 @@ TryToDelegateFunctionCall(DistributedPlanningContext *planContext)
 	{
 		/*
 		 * We are planning a call to a distributed function within a Citus backend,
-		 * that means that this is the delegated call.
+		 * that means that this is the delegated call. If the function is forcefully
+		 * delegated, capture the distribution argument.
 		 */
-		InDelegatedFunctionCall = true;
+		if (procedure->forceDelegation)
+		{
+			CheckDelegatedFunctionExecution(procedure, funcExpr);
+		}
+
+		/* Are we planning the top function call? */
+		if (ExecutorLevel == 1 || PlannerLevel == 1)
```

Comment:
do we care about ExecutorLevel here? (we are normally in the planner only, if the executor invokes the planner, it suggests we are not at the top level)

Reply:
btw, it seems like this could be:
```
else if (PlannerLevel == 1 && !IsMultiStatementTransaction()))
{
    /* we are planning a regular delegated call */
    InTopLevelDelegatedFunctionCall = true;
}
```

Reply:
That would miss the scenario of forceDelegation == true && !IsMultiStatementTransaction(), which is allowed to do remote execution

Reply:
As discussed, changed this to if (ExecutorLevel == 0 && PlannerLevel == 1)